src/command/Command.php

```diff
@@ -112,8 +112,9 @@ public function testPermission(CommandSender $target, ?string $permission = null
 			return true;
 		}
 
-		if($this->permissionMessage === null){
-			$target->sendMessage(KnownTranslationFactory::pocketmine_command_error_permission($this->name)->prefix(TextFormat::RED));
+		if($this->permissionMessage instanceof Translatable || $this->permissionMessage === null){
```

Comment:
This logic looks cursed. It would be better to move the message variable outside the condition and use it in all the branches.

Reply:
Huh ? What do you mean by moving outside the conditions? I don't think I understand then.